controllers/etcdcluster_controller.go

```diff
@@ -36,7 +36,7 @@ type EtcdClusterReconciler struct {
 	client.Client
 	Log      logr.Logger
 	Recorder record.EventRecorder
-	Etcd     etcd.EtcdAPI
+	Etcd     etcd.APIBuilder
```

Comment:
Because previously the method was `MembershipAPI`, which *also* created the client and the connection.
Now I've added a `GetVersion` method, which is not part of the Etcd `MembersAPI` it seemed appropriate to use  this name (`EtcdAPI`) for the interface having all the Etcd methods that we actually use in the operator.
So I introduced `APIBuilder` as the interface we use in the operator to establish a connection to one one of the clusters.